includes/options/class-amp-options-manager.php

```diff
@@ -286,6 +287,17 @@ public static function validate_options( $new_options ) {
 			unset( $options[ Option::DISABLE_CSS_TRANSIENT_CACHING ] );
 		}
 
+		/**
+		 * Filter the options being updated, so services can handle the sanitization and validation of
+		 * their respective options.
+		 *
+		 * @internal
+		 *
+		 * @param array $options     Existing options with already-sanitized values for updating.
+		 * @param array $new_options Unsanitized options being submitted for updating.
+		 */
+		$options = apply_filters( 'amp_options_updating', $options, $new_options );
```

Comment:
@schlessera I've added this to allow the option updating logic to be removed from the `AMP_Options_Manager` and over to the respective services.

Reply:
I thought about turning the individual options into objects to inject as dependencies already. But the filter will do the trick for now.

BTW, this is what I referred to when I meant that we might need to add hooks so we can reverse the flow of the logic. So we seem to be on the same page wrt this.